src/toil/fileStores/cachingFileStore.py

```diff
@@ -97,1172 +113,1457 @@ class CachingFileStore(AbstractFileStore):
-        else:
-            jobStoreFileID = self.jobStore.writeFile(absLocalFileName, creatorID, cleanup)
-            # Non local files are NOT cached by default, but they are tracked as local files.
-            self._JobState.updateJobSpecificFiles(self, jobStoreFileID, None,
-                                                  0.0, False)
-        return FileID.forPath(jobStoreFileID, absLocalFileName)
 
-    def writeGlobalFileStream(self, cleanup=False):
-        # TODO: Make this work with caching
-        return super(CachingFileStore, self).writeGlobalFileStream(cleanup)
+        # Get a cursor
+        cur = con.cursor()
+
+        cur.execute("""
+            CREATE TABLE IF NOT EXISTS files (
+                id TEXT NOT NULL PRIMARY KEY, 
+                path TEXT UNIQUE NOT NULL,
+                size INT NOT NULL,
+                state TEXT NOT NULL,
+                owner INT 
+            )
+        """)
+        cur.execute("""
+            CREATE TABLE IF NOT EXISTS refs (
+                path TEXT NOT NULL,
+                file_id TEXT NOT NULL,
+                job_id TEXT NOT NULL,
+                state TEXT NOT NULL,
+                PRIMARY KEY (path, file_id)
+            )
+        """)
+        cur.execute("""
+            CREATE TABLE IF NOT EXISTS jobs (
+                id TEXT NOT NULL PRIMARY KEY,
+                tempdir TEXT NOT NULL,
+                disk INT NOT NULL,
+                worker INT
+            )
+        """)
+        cur.execute("""
+            CREATE TABLE IF NOT EXISTS properties (
+                name TEXT NOT NULL PRIMARY KEY,
+                value INT NOT NULL
+            )
+        """)
+        con.commit()
+    
+    # Caching-specific API
 
-    def readGlobalFile(self, fileStoreID, userPath=None, cache=True, mutable=False, symlink=False):
+    def getCacheLimit(self):
         """
-        Downloads a file described by fileStoreID from the file store to the local directory.
-        The function first looks for the file in the cache and if found, it hardlinks to the
-        cached copy instead of downloading.
-        The cache parameter will be used only if the file isn't already in the cache, and
-        provided user path (if specified) is in the scope of local temp dir.
-        :param bool cache: If True, a copy of the file will be saved into a cache that can be
-               used by other workers. caching supports multiple concurrent workers requesting the
-               same file by allowing only one to download the file while the others wait for it to
-               complete.
-        :param bool mutable: If True, the file path returned points to a file that is
-               modifiable by the user. Using False is recommended as it saves disk by making
-               multiple workers share a file via hard links. The default is False.
+        Return the total number of bytes to which the cache is limited.
+
+        If no limit is in place, return None.
```

Comment:
Maybe. This is supposed to mirror the old logic, and the way space is accounted kind of revolves around having job space and cache space live in the same pool. We could say that only a certain maximum fraction of that space is allowed to be used for the cache, but then we would do weird stuff like write a file immediately instead of caching it by hardlink and writing it later because the cache didn't have room to take ownership of the file's space usage from the job.